### DISCUSSION TITLE
How to create plugins on top of Ant Media Server?

### DISCUSSION THREAD

**Discussion by USAMAWIZARD** [ant-media/Ant-Media-Server#5004] [FAQs] [answered]
We would like to integrate our extensions with Ant Media Server using plugins.
How can we do that and build plugins on top of Ant Media Server?

**Comment by Mohit-3196** [comment, marked as answer]
Ant Media Server supports plugins to integrate them into the video/audio processing in it. You can have a plugin that captures video/audio packets (encoded) or frames (decoded) data from Ant Media Server. Then your plugin can consume the data or process them and feed them back to the Ant Media Server.

You can check this [blogpost](https://antmedia.io/plugins-will-make-ant-media-server-more-powerful/) to understand the plugin architecture and check [this](https://antmedia.io/plugin-development-guide/) to learn how you can create a plugin.

Also, I recommend you to check Ant Media's Plugins [repository](https://github.com/ant-media/Plugins). There are open-source plugins developed by the team.

Thank you.
